mmaction/datasets/pipelines/augmentations.py

```diff
@@ -391,132 +394,99 @@ def __repr__(self):
         return repr_str
 
 
-# Note, entity box transfroms are not added to: ThreeCrop, TenCrop,
-# MultiGroupCrop.
-@PIPELINES.register_module()
-class EntityBoxRescale:
+# box ops are not added to: ThreeCrop, TenCrop, MultiGroupCrop.
+def box_resize(results, scale_factor):
     """Rescale the entity box and proposals according to the image shape.
 
-    Required keys are "proposals", "gt_bboxes", added or modified keys are
-    "gt_bboxes". If original "proposals" is not None, "proposals" and
-    will be added or modified.
+    Required keys in results are "proposals", "gt_bboxes", added or modified
+    keys are "gt_bboxes" and "proposals" (if not None).
 
     Args:
+        results (dict): It contains all information of a data sample.
         scale_factor (np.ndarray): The scale factor used entity_box rescaling.
     """
 
-    def __init__(self, scale_factor):
-        self.scale_factor = scale_factor
-
-    def __call__(self, results):
-        scale_factor = np.concatenate([self.scale_factor, self.scale_factor])
-
-        proposals = results['proposals']
-        gt_bboxes = results['gt_bboxes']
-        results['gt_bboxes'] = gt_bboxes * scale_factor
+    assert len(scale_factor) == 2
+    scale_factor = np.concatenate([scale_factor, scale_factor])
+    assert 'proposals' in results and 'gt_bboxes' in results
+    results['gt_bboxes'] = results['gt_bboxes'] * scale_factor
+    if results['proposals'] is not None:
+        assert results['proposals'].shape[1] == 4
+        results['proposals'] = results['proposals'] * scale_factor
+    return results
 
-        if proposals is not None:
-            assert proposals.shape[1] == 4, (
-                'proposals shape should be in '
-                f'(n, 4), but got {proposals.shape}')
-            results['proposals'] = proposals * scale_factor
 
-        return results
-
-    def __repr__(self):
-        return f'{self.__class__.__name__}(scale_factor={self.scale_factor})'
-
-
-@PIPELINES.register_module()
-class EntityBoxCrop:
+def box_crop(results, crop_bbox):
```

Comment:
Since `EntityBoxCrop` etc. are exported as part of our interface, so we need to add deprecation warnings first, and then remove them.

You can make a dummy implementation of `EntityBoxCrop` (with dep warn) by using the new `box_crop` function.

Reply:
That's OK. Besides, although these classes are provided, they are never used alone in the pipeline,